signer/src/storage/sqlx.rs

```diff
@@ -330,3 +343,68 @@ impl sqlx::postgres::PgHasArrayType for SigHash {
         <[u8; 32] as sqlx::postgres::PgHasArrayType>::array_type_info()
     }
 }
+
+// --- sqlx Type implementation for Timestamp --
+
+impl sqlx::Type<sqlx::Postgres> for Timestamp {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::with_oid(TIMESTAMPTZ_OID)
+    }
+
+    fn compatible(ty: &PgTypeInfo) -> bool {
+        // Ensure compatibility with PostgreSQL's TIMESTAMPTZ type.
+        ty.oid() == Some(TIMESTAMPTZ_OID)
+    }
+}
+
+impl<'q> sqlx::Encode<'q, sqlx::Postgres> for Timestamp {
+    fn encode_by_ref(
+        &self,
+        buf: &mut sqlx::postgres::PgArgumentBuffer,
+    ) -> Result<IsNull, BoxDynError> {
+        let unix_seconds = self.unix_timestamp();
+
+        // Convert Unix seconds to microseconds since PostgreSQL epoch.
+        let pg_epoch_micros = (unix_seconds - PG_EPOCH_SECONDS_FROM_UNIX_EPOCH)
+            .checked_mul(MICROS_PER_SECOND)
+            .ok_or_else(|| {
+                sqlx::Error::Encode("timestamp value out of range for postgres TIMESTAMPTZ".into())
+            })?;
+
+        pg_epoch_micros.encode_by_ref(buf)
+    }
+
+    fn size_hint(&self) -> usize {
+        std::mem::size_of::<i64>()
+    }
+}
+
+impl<'r> sqlx::Decode<'r, sqlx::Postgres> for Timestamp {
+    fn decode(value: sqlx::postgres::PgValueRef<'r>) -> Result<Self, BoxDynError> {
+        // Decode the i64 representing microseconds since PostgreSQL epoch.
+        let pg_epoch_micros = <i64 as sqlx::Decode<sqlx::Postgres>>::decode(value)?;
+
+        // Convert microseconds since PostgreSQL epoch to seconds since Unix
```

Comment:
Do we need to worry about the format of the value returned from postgres? The sqlx implementation suggests that we should, but I'm not sure. https://github.com/launchbadge/sqlx/blob/bab1b022bd56a64f9a08b46b36b97c5cff19d77e/sqlx-postgres/src/types/time/datetime.rs#L54-L89

Reply:
Not according to the integration test, anyway. It looks like their implementation handles conversion from textual format, etc. We could always add that, but `TIMESTAMPZ` itself should be stored as an `i64` iiuc.

Reply:
It's weird that the sqlx folks support whatever it is that they support there; it's as if they support going from a varchar to a timestamp, but that cannot be what they are actually supporting. Hmmm, seems like it's something to keep in mind.

Reply:
Resolving since there isn't really anything actionable.

Reply:
Yeah, but I find that so strange. Wouldn't the OID read differently for TEXT data, or something like that? Well I don't understand how it all works so maybe it makes some sense.